codex-rs/core/src/codex.rs

```diff
@@ -1469,8 +1483,9 @@ async fn handle_container_exec_with_params(
     sess.notify_exec_command_begin(&sub_id, &call_id, &params)
         .await;
 
+    let processed_params = maybe_run_with_shell(params.clone(), sess);
```

Comment:
Consider just shadowing, which avoids `clone`?

```suggestion
    let params = maybe_run_with_shell(params.clone(), sess);
```